content/docs/development/health-rules.md

```diff
@@ -0,0 +1,168 @@
+# Health Rules
+
+Cartographer reports the health of each object stamped on the cluster. If no health rule is specified in the template,
+Cartographer's health behavior will be rudimentary:
+
+- if the object is rejected by the API server (e.g. if you mistakenly try to template out a CinfogMap instead of a
+  ConfigMap) then the workload will report the resource's Healthy condition as "Unknown" for the reason
+  "OutputNotAvailable". The workload will report the resource's Ready condition as "False" for the reason
```

Comment:
```suggestion
  "NoStampedObject". The workload will report the resource's Ready condition as "False" for the reason
```

I think the reason mentioned here is outdated as, applying the following

```yaml
---
apiVersion: carto.run/v1alpha1
kind: Workload
metadata:
  name: app
  labels:
    foo: bar
spec: {}


---
apiVersion: carto.run/v1alpha1
kind: ClusterSupplyChain
metadata:
  name: app
spec:
  selector:
    foo: bar

  resources:
    - name: first
      templateRef:
        kind: ClusterTemplate
        name: first


---
apiVersion: carto.run/v1alpha1
kind: ClusterTemplate
metadata:
  name: first
spec:
  ytt: |
    #@ load("@ytt:data", "data")
    ---
    apiVersion: v1
    kind: ConfssssigMap
    metadata:
      name: #@ data.values.workload.metadata.name
    data:
      foo: bar
```

leads to

```yaml
status:
  resources:
    conditions:
    - lastTransitionTime: "2022-08-19T18:59:42Z"
      message: ""
      reason: NoStampedObject
      status: Unknown
      type: Healthy

```

Reply:
Looks like it depends on whether you are using a `ClusterTemplate` or not.

```yaml
---
apiVersion: carto.run/v1alpha1
kind: ClusterImageTemplate
metadata:
  name: image-template
spec:
  imagePath: .data.average_color_of_the_universe
  template:
    apiVersion: v1
    kind: ConfigureMap
    metadata:
      name: test-configmap-image-1
    data:
      foo: bar

---
apiVersion: carto.run/v1alpha1
kind: ClusterTemplate
metadata:
  name: bare-template
spec:
  template:
    apiVersion: v1
    kind: ConfigureMap
    metadata:
      name: test-configmap-image-2
    data:
      foo: bar

---
apiVersion: carto.run/v1alpha1
kind: ClusterSupplyChain
metadata:
  name: app
spec:
  selector:
    foo: bar

  resources:
    - name: first
      templateRef:
        kind: ClusterImageTemplate
        name: image-template
    - name: second
      templateRef:
        kind: ClusterTemplate
        name: bare-template

---
apiVersion: carto.run/v1alpha1
kind: Workload
metadata:
  name: app
  labels:
    foo: bar
spec: {}
```

when you look at the resources' health condition, it is different for the object stamped by a ClusterImageTemplate and the object stamped by the ClusterTemplate.

```console
kubectl get -o yaml workload app | yq '.status.resources[] | .conditions[1]'
```

``` json
{
  "lastTransitionTime": "2022-08-19T20:18:05Z",
  "message": "",
  "reason": "OutputNotAvailable",
  "status": "Unknown",
  "type": "Healthy"
}
{
  "lastTransitionTime": "2022-08-19T20:18:08Z",
  "message": "",
  "reason": "NoStampedObject",
  "status": "Unknown",
  "type": "Healthy"
}
```

Reply:
I simply removed the reference to what the `Reason` will be, since it can change depending on the template.